internal/storage/streamed_chunk_uploader.go

```diff
@@ -0,0 +1,206 @@
+package storage
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	"cloud.google.com/go/storage"
+	"github.com/googlecloudplatform/gcsfuse/internal/logger"
+	"github.com/googlecloudplatform/gcsfuse/internal/storage/gcs"
+	"github.com/googlecloudplatform/gcsfuse/internal/storage/storageutil"
+	"google.golang.org/api/googleapi"
+)
+
+type UploaderState int
+
+const (
+	// Default initial state in case of failure in initialization.
+	Uninitialized UploaderState = iota
+	// State of successful initialization, no writes so far.
+	Initialized
+	// No write is in progress, but there have been writes in the past.
+	Waiting
+	// Asynchronous write is in progress.
+	Writing
+	// Upload failed.
+	WriteError
+	// Has been finalized.
+	Destroyed
+)
+
+// A StreamedChunkUploader is an implementation of ChunkUploader
+// interface, which use storage.Writer from go-storage-client
+// for resumable upload.
+//
+// It also stores state of how many bytes have been sent to it
+// to write, how many have been successfully written by it so far,
+// and the current state of the uploader.
+type StreamedChunkUploader struct {
+	// Implements the following interfaces.
```

Comment:
If the struct implements all the interface methods, it is assumed that struct implements the interface. no need of this line.

Reply:
@vadlakondaswetha Do you mean just the comment or the interface declaration on the next line as well is not needed? 

I know the interface declaration isn't necessary, but I've made it explicit to make it easier for the reader, e.g. in https://github.com/GoogleCloudPlatform/gcsfuse/blob/66b8100310ff701a7f4fba5f801615c3480228dc/internal/gcsx/content_type_bucket.go#L32 .

Reply:
The reason for adding inside content_type_bucket struct is to store the reference of original implementation and call it in other methods:

https://github.com/GoogleCloudPlatform/gcsfuse/blob/66b8100310ff701a7f4fba5f801615c3480228dc/internal/gcsx/content_type_bucket.go#L28
https://github.com/GoogleCloudPlatform/gcsfuse/blob/66b8100310ff701a7f4fba5f801615c3480228dc/internal/gcsx/content_type_bucket.go#L44

 I dont think there is any other definition which uploadChunker wants to call.

Reply:
Ok. Removed.